File: app/src/main/java/org/odddev/fantlab/home/NavDrawerRouter.java
package org.odddev.fantlab.home;

import android.support.annotation.IdRes;
import android.support.annotation.IntDef;

import org.odddev.fantlab.R;
import org.odddev.fantlab.auth.AuthActivity;
import org.odddev.fantlab.award.AwardsFragment;
import org.odddev.fantlab.catalog.CatalogFragment;
import org.odddev.fantlab.core.layers.router.Router;
import org.odddev.fantlab.core.utils.FragmentUtils;
import org.odddev.fantlab.profile.ProfileFragment;
import org.odddev.fantlab.search.SearchFragment;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

/**
 * @author kenrube
 * @since 11.10.16
 */

class NavDrawerRouter extends Router<HomeActivity> {

    @IntDef({
            R.id.nav_autors,
            R.id.nav_awards,
            R.id.nav_search,
            R.id.nav_profile,
            R.id.nav_logout,
            R.id.nav_login
    })
    @Retention(RetentionPolicy.SOURCE)
    @interface NAV_DRAWER_ITEM{}

    NavDrawerRouter(HomeActivity activity, @IdRes int containerId) {
        super(activity, containerId);
    }

    boolean routeToNavDrawerItem(@NAV_DRAWER_ITEM int item) {
        switch (item) {
            case R.id.nav_autors: {
                FragmentUtils.replaceFragment(getActivity(), getContainerId(), new CatalogFragment(), false);
                return true;
            }
            case R.id.nav_awards: {
                FragmentUtils.replaceFragment(getActivity(), getContainerId(), new AwardsFragment(), false);
                return true;
            }
            case R.id.nav_search: {
                FragmentUtils.replaceFragment(getActivity(), getContainerId(), new SearchFragment(), false);
                return true;
            }
            case R.id.nav_profile: {
                FragmentUtils.replaceFragment(getActivity(), getContainerId(), new ProfileFragment(), false);
                return true;
            }
            case R.id.nav_logout:
            case R.id.nav_login: {
                AuthActivity.start(getActivity());
                getActivity().overridePendingTransition(0, 0);
                return true;
            }
        }
        return false;
    }
}

File: app/src/main/kotlin/org/odddev/fantlab/home/NavDrawerRouter.kt
package org.odddev.fantlab.home

import android.annotation.SuppressLint
import android.content.Intent
import android.support.annotation.IdRes
import android.support.annotation.IntDef
import org.odddev.fantlab.R
import org.odddev.fantlab.auth.AuthActivity
import org.odddev.fantlab.authors.AuthorsFragment
import org.odddev.fantlab.core.layers.router.Router
import org.odddev.fantlab.core.utils.FragmentUtils
import org.odddev.fantlab.search.SearchFragment

internal class NavDrawerRouter(activity: HomeActivity, @IdRes containerId: Int) : Router<HomeActivity>(activity, containerId) {

	@IntDef(R.id.nav_authors.toLong()/*, R.id.nav_awards.toLong()*/, R.id.nav_search.toLong(),
			/*R.id.nav_profile.toLong(),*/ R.id.nav_logout.toLong(), R.id.nav_login.toLong())
	@Retention(AnnotationRetention.SOURCE)
	internal annotation class NAV_DRAWER_ITEM

	@SuppressLint("SwitchIntDef")
	fun routeToNavDrawerItem(@NAV_DRAWER_ITEM item: Int) {
		when (item) {
			R.id.nav_authors -> FragmentUtils.replaceFragment(activity, containerId, AuthorsFragment(), false)
			R.id.nav_search -> FragmentUtils.replaceFragment(activity, containerId, SearchFragment(), false)
			//R.id.nav_awards -> FragmentUtils.replaceFragment(activity, containerId, AwardsFragment(), false)
			R.id.nav_login ->
				run {
					activity.startActivity(Intent(activity, AuthActivity::class.java))
					activity.finish()
				}
			R.id.nav_logout ->
				run {
					activity.startActivity(Intent(activity, AuthActivity::class.java))
					activity.finish()
				}
		}
	}
}
